tools/c7n_azure/c7n_azure/resources/subscription.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2016-2017 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import six
+
+from azure.mgmt.resource.policy.models import PolicyAssignment
+from azure.mgmt.subscription import SubscriptionClient
+
+from c7n.actions import BaseAction
+from c7n.exceptions import PolicyValidationError
+from c7n.manager import ResourceManager
+from c7n.utils import local_session, type_schema
+
+from c7n_azure.provider import resources
+from c7n_azure.query import QueryMeta
+
+
+@resources.register('subscription')
+@six.add_metaclass(QueryMeta)
+class Subscription(ResourceManager):
+
+    class resource_type(object):
+        id = 'subscriptionId'
+        name = 'displayName'
+        filter_name = None
+
+    def get_model(self):
+        return self.resource_type
+
+    def resources(self):
+        return self.filter_resources([self._get_subscription(self.session_factory, self.config)])
+
+    def get_resources(self, resource_ids):
+        return [self._get_subscription(self.session_factory, self.config)]
+
+    def _get_subscription(self, session_factory, config):
+        session = local_session(session_factory)
+        client = SubscriptionClient(session.get_credentials())
+        details = client.subscriptions.get(subscription_id=session.subscription_id)
+        return {'subscriptionId': session.subscription_id,
+                'displayName': details.display_name}
+
+
+@Subscription.action_registry.register('add-policy')

```

Comment:
normally we try to do set-* instead of add/remove-* to allow for idempotent interaction and limiting the vocabulary of things users have to learn around a feature/functionality.

Reply:
perhaps just do set-policy / state: present for now and can implement state: absent case in future.

Reply:
discussed in person, delete on policy binding is already present as policy binding is a separate resource, this particular use case is in a negative assertion scenario, the result will be convergent on name.